dask/dataframe/io/io.py

```diff
@@ -562,12 +566,20 @@ def from_delayed(dfs, meta=None, divisions=None, prefix="from-delayed"):
 
     name = prefix + "-" + tokenize(*dfs)
     dsk = merge(df.dask for df in dfs)
-    dsk.update(
-        {
-            (name, i): (check_meta, df.key, meta, "from_delayed")
-            for (i, df) in enumerate(dfs)
-        }
-    )
+    if check_meta_flag:
+        dsk.update(
+            {
+                (name, i): (check_meta, df.key, meta, "from_delayed")
+                for (i, df) in enumerate(dfs)
+            }
+        )
+    else:
+        dsk.update(
+            {
+                (name, i): (lambda x, *_: x, df.key, meta, "from_delayed")
```

Comment:
This lambda seems to be a no-op.  Maybe insteaad something like the following?

```suggestion
                (name, i): df.key,
```